.github/workflows/ci.yml

```diff
@@ -29,32 +29,24 @@ jobs:
       with:
         elixir-version: ${{matrix.elixir}}
         otp-version: ${{matrix.otp}}
-    - name: Restore dependencies cache
-      id: restore-deps
-      uses: actions/cache@v4
-      with:
-        path: deps
-        key: deps-${{ runner.os }}-${{ matrix.otp }}-${{ matrix.elixir }}-${{ hashFiles('**/mix.lock') }}
-        restore-keys: deps-${{ runner.os }}-${{ matrix.otp }}-${{ matrix.elixir }}-
-    - name: Install dependencies
-      run: mix deps.get --only test
-    - name: Restore build cache
+    - name: Cache dependencies
+      id: cache-deps
       uses: actions/cache@v4
       with:
         path: |
           _build
           deps
-          ~/.hex
-          ~/.mix
-        key: build-${{ runner.os }}-${{ matrix.otp }}-${{ matrix.elixir }}-${{ hashFiles('**/mix.lock') }}
-        restore-keys: build-${{ runner.os }}-${{ matrix.otp }}-${{ matrix.elixir }}-
+        key: ${{ runner.os }}-${{ matrix.otp }}-${{ matrix.elixir }}-${{ hashFiles('**/mix.lock') }}
+        restore-keys: ${{ runner.os }}-${{ matrix.otp }}-${{ matrix.elixir }}-
+    - name: Install and compile dependencies
+      run: mix do deps.get --only test, deps.compile
```

Comment:
💭 This step could conceivably be skipped if the cache was hit in the previous `cache-deps` step.

Reply:
Will try that now.